Add EnumOption for SpeciesSite::SiteType

rprospero opened the pull request:
This closes #1706

rprospero committed:
Add EnumOption for SpeciesSite::SiteType

trisyoungs commented on src/classes/speciesSite.h:
```diff
@@ -25,12 +25,15 @@ class SpeciesSite : public Serialisable<CoreData &>
         Fragment /* Site is based on a NETA description */
     };
+    // EnumOptions for SiteType
+    static EnumOptions<SiteType> types();
```
To be consistent with the rest of the code:
```suggestion
    static EnumOptions<SiteType> siteTypes();
```

trisyoungs commented on src/classes/speciesSite.cpp:
```diff
@@ -23,6 +23,13 @@ SpeciesSite::SpeciesSite(const Species *parent, std::string name, SiteType type)
+EnumOptions<SpeciesSite::SiteType> SpeciesSite::types()
+{
+    return EnumOptions<SiteType>(
+        "SiteType", {{SiteType::Static, "static"}, {SiteType::Dynamic, "dynamic"}, {SiteType::Fragment, "fragment"}});
```
Now, here's an annoying point for discussion: historically all enum strings had first letter capitalised so that the inputfile (.txt) looked "nice", but of course moving away from that we don't really care, and the lowercase first letter is of course more consistent with the TOML styling we're moving towards.  However, as just discussed in the stand-up, capitalising enums to distinguish them from variables is a sound way forward!

trisyoungs reviewed (approved):
Couple of comments, as discussed.

:+1:

rprospero committed:
Make enum strings upper-case

It's a good convention in the TOML that enums are upper case and
variables are lower case.

rprospero committed:
Better method name for siteTypes()
